src/librustc_mir/borrow_check/mod.rs

```diff
@@ -1465,6 +1501,12 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
                         // FIXME: is this true even if P is a adt with a dtor?
                         { }
 
+                        ProjectionElem::Deref => {
+                            self.check_if_path_is_moved(
+                                context, InitializationRequiringAction::Use,
+                                (base, span), flow_state);
```

Comment:
You can add a `break` here if you want - we already force the base to be initialized, so there is no need to further check.

Maybe in that case you would also not need the new error deduplication code.